scrapy/pipelines/files.py

```diff
@@ -261,13 +270,19 @@ def from_settings(cls, settings):
         store_uri = settings['FILES_STORE']
         return cls(store_uri, settings=settings)
 
-    def _get_store(self, uri):
+    def _get_store(self, uri, settings):
         if os.path.isabs(uri):  # to support win32 paths like: C:\\some\dir
             scheme = 'file'
         else:
             scheme = urlparse(uri).scheme
         store_cls = self.STORE_SCHEMES[scheme]
-        return store_cls(uri)
+        store = store_cls(uri)
+        # this is a bit ugly but since S3FilesStore does not accept
+        # settings when instantiated, and to keep it backward-ompatible,
+        # the store object is patched with the region name
+        if scheme == 's3':
```

Comment:
@vshlapakov related: https://github.com/scrapy/scrapy/pull/2044 and https://github.com/scrapy/scrapy/pull/1605